src/table.js

```diff
@@ -538,15 +543,100 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
+  const numberDefault = defaultValue === null ? NaN : defaultValue;
+  const stringValue = typeof value === "string" && !options.soft ? value.trim() : value;
+  switch (type) {
+    case "string": 
+      return typeof value === "string"
+        ? stringValue
+        : value || value === 0
+        ? value.toString()
+        : defaultValue;
+    case "boolean":
+      return value === true || stringValue === "true"
+        ? true
+        : value === false || stringValue === "false"
+        ? false
+        : defaultValue;
+    case "integer":
+      return !value || isNaN(parseInt(value)) ? numberDefault : parseInt(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? numberDefault
+        // eslint-disable-next-line no-undef
+        : BigInt(value);
+    case "number": {
+      return !value || isNaN(value) ? numberDefault : Number(value);
```

Comment:
This `!value` also results in returning `NaN` for the number `0`. Seems like we should add a test that includes the number `0`.